docs/development/CODE_REVIEW_GUIDELINES.md

```diff
@@ -0,0 +1,121 @@
+
+For decades, the IETF (Internet Engineering Task Force) has used the motto "rough consensus and running code".
+This motto encapsulates ("running code") that developers' core output is still software: if there is no code that runs and produces correct results, we have nothing.
+It also encapsulates ("rough consensus") that we may not always precisely agree and that's okay.
+
+## Code Reviews
+
+One tactic often employed to produce "maintainable" or "more-readable" code is a "code review".
+These can take many shapes and forms and often have goals beyond simply "code quality".
+
+Broadly speaking, code reviews involve developers looking at some proposed new code (or code changes).
+This is *usually* developers other than the author (although a "self-review" can also be employed).
+In many projects, such attention is a scarce commodity: most programmers would rather write code than read someone else's.
+
+## Code Review Guidelines
+
+All participants must adhere to the overall idea that the review is an attempt to achieve "better" code.
+This is a vague statement on purpose.
+Participants must be cautious in their criticism and generous with praise.
+Participants must remember the scarcity of another developer's attention.
+
+### Expectations
+
+**Pull Request Authors:**
+The author is responsible for tracking up-to-date next actions for a Pull Request to progress towards being merged.
+
+**Reviewers:**
+Reviewers should prefer engaging in code review over starting new work (i.e. taking planned work items that haven't been started yet).
+
+Reviewers (and prospective reviewers) are encouraged to engage in reviews of codebases outside the projects and technologies they use on a day-to-day basis (but not expected to provide an approving review).
+
+## Best Practices
+
+### Reviewing
+
+### Starting a Review
+
+Consider if it could be broken into smaller Pull Requests. If it is clear that it can be, summarize your thinking on how in your Review.
+
+If the commits be (re)organized (i.e. reordered and/or amended) such that there is a commit at which the tests are passing prior to the conclusion of the main change, that's a signal that there's likely a logic split which can be made at that point in such a (re)arrangement.
+
+Use the following guidelines to evaluate whether a Pull Request should be approved:
+
+- *Every* Pull Request should have tests or justification otherwise; esp. bug fixes.
+- *Every* Pull Request should have at least 1 approval from a team member internal or external to the project. Exceptions made by repository maintainer(s), as necessary, on a case-by-case basis.
+
+During review, submit feedback using line comments (A); prefer "Add\[ing a\] single comment" over "Start[ing] a review" (B).
+
+*(Once a review has been started, the option to add single comments is removed. Preferring single comments allows feedback to happen even in the event of an interrupted review.)*
```

Comment:
I personally prefer Starting reviews because I often go back and edit/remove comments if my question/comment was addressed elsewhere.

Is this what you found yourself doing recently?

Reply:
> I personally prefer Starting reviews because I often go back and edit/remove comments if my question/comment was addressed elsewhere.
> 
Same here. Review comments stay pending until you finish the review, so I can leave placeholder comments on items that may need a change request. It is very common to encounter something else further down the diff that establishes a previously suspicious item as correct.

Reply:
> Is this what you found yourself doing recently?

I've been doing it this way for several years now. My experience has been that I may not have time to run down every thread in one review session but more often than not review comments / suggestions are independently actionable and non-blocking with respect to one another. The intention was to increase team velocity by not withholding already actionable feedback.

I acknowledge and can also relate the case where you aren't sure about a comment just yet and want to wait until you're sure you've fully groked it or ran down / validated a related concern. I've chosen to prefer editing a premature comment over having more than a couple of comments pending at once. With only a couple of comments pending, I've found it pretty reasonable to leave the text in the unsubmitted textarea and/or copy things between there and my notes for larger scope feedback / investigations.

Reply:
I'm realizing a couple of other hangups that I have with the combined review comment UX:

1. It's likely to cause duplicate comments in the PR timeline
2. The review comment itself and related comments can't be "resolved" (only supported action is limite emoji reaction)

I threw together a quick [demo PR on my fork](https://github.com/bryanchriswhite/pocket/pull/8) where you can see that the third reviews comments are duplicated and unresolvable in the combined review copy.